kotlin/samples/todo-android/common/src/main/java/com/squareup/sample/todo/MasterDetailAware.kt

```diff
@@ -0,0 +1,62 @@
+package com.squareup.sample.todo
+
+/**
+ * Informs [MasterDetailAware] renderings whether they're children of a [MasterDetailScreen],
+ * and if so in what configuration.
+ */
+enum class MasterDetailConfig {
+  /**
+   * Parent is not a [MasterDetailScreen]. Expected to be used as the default value
+   * of [MasterDetailAware.masterDetailConfig].
+   */
+  None,
+
+  /**
+   * Rendering is in [MasterDetailScreen.masterRendering], running in split screen
+   * configuration.
+   */
+  Master,
+
+  /**
+   * Rendering is in [MasterDetailScreen.detailRendering], running in split screen
+   * configuration.
+   */
+  Detail,
+
+  /**
+   * Rendering is the only visible child in a [MasterDetailScreen] running in
+   * single screen configuration.
+   */
+  Only
+}
+
+/**
+ * Implemented by rendering types to allow them to customize appearance / behavior
+ * based on a [MasterDetailScreen] parent's [configuration][MasterDetailConfig].
+ */
+interface MasterDetailAware<T : MasterDetailAware<T>> {
+  /**
+   * The configuration of a possible [MasterDetailScreen] parent rendering.
+   * Should default to [MasterDetailConfig.None].
+   */
+  val masterDetailConfig: MasterDetailConfig
```

Comment:
Thought about making this `MasterDetailConfig?` and getting rid of the `None` value (and same in back stack), but didn't seem like it would make anything more readable.

Reply:
What happens if you default to another value?

Reply:
Then you're lying to yourself. If you get rendered in a container other than `MasterDetailScreen` there is no promise that the value will be updated.